### ISSUE TITLE
ClassCastException on simple search

### ISSUE THREAD

**Issue by dhelleberg** [adamint/spotify-web-api-kotlin#26]
Hi, 
I tried a simple track search according to the documentation and example files, but I'm getting a ClassCast Exception when calling the search method. 
What is wrong with my code?
I profiled the network an the REST Calls / Response seems to be OK

```
val result = api.search.searchTrack("quand je serai grand", market = Market.FR).complete();
println(result.items)
```
throws this:

```
java.lang.ClassCastException: org.json.JSONArray cannot be cast to java.lang.Iterable
     at com.adamratzman.spotify.endpoints.pub.search.SearchAPI$searchTrack$1.get(SearchAPI.kt:96)
     at com.adamratzman.spotify.endpoints.pub.search.SearchAPI$searchTrack$1.get(SearchAPI.kt:10)
     at com.adamratzman.spotify.utils.SpotifyRestAction.complete(SpotifyRestAction.kt:11)
```

**Comment by adamint** [comment]
Hi @dhelleberg, thanks for bringing this up.
Could you do me a favor and use the following for today?

```
repositories {
	maven { url 'https://jitpack.io' }
}
```
(add that to your build.gradle repositories block)
 
and 
```
dependencies {
	compile 'com.github.adamint:spotify-web-api-kotlin:dev-SNAPSHOT'
}
```

The issue is that the old endpoint has been replaced, and I won't be releasing 2.0.0 until later tonight.

**Comment by molikuner** [comment]
Although this just works for me (tested 5 min ago with version 1.0.3), it's probably the best idea.

**Comment by adamint** [comment]
I'm actually staging `2.0.0` right now

**Comment by adamint** [comment]
Okay, please retry using `2.0.0`

**Comment by dhelleberg** [comment]
Thanks a lot for the quick reply, I tried 2.0.0 but I have similar issues. Here's my complete code I use from the lib and the according exception. I run on android using kotlin 1.3.10 and I'm running the code below from a coroutine.


```
val api = spotifyApi {
             credentials {
                 clientId = CLIENT_ID
                 clientSecret = CLIENT_SECRET
             }
         }.buildCredentialed()
         try {
             // block and print out the names of the twenty most similar songs to the search
             api.search.searchTrack("Début de la Suite").complete().map { it.name }.joinToString().let { Log.d(TAG,it) }


         } catch (e: Exception) {
             Log.e(TAG,"Oh no! Error: ${e.printStackTrace()}",e)
         }

```
```
Exception:
java.lang.ClassCastException: org.json.JSONArray cannot be cast to java.lang.Iterable
     at com.adamratzman.spotify.utils.HelpersKt.toPagingObject(Helpers.kt:150)
     at com.adamratzman.spotify.endpoints.public.SearchAPI$searchTrack$1.get(SearchAPI.kt:127)
     at com.adamratzman.spotify.endpoints.public.SearchAPI$searchTrack$1.get(SearchAPI.kt:22)
     at com.adamratzman.spotify.main.SpotifyRestAction.complete(SpotifyRestAction.kt:12)
     at com.adamratzman.spotify.main.SpotifyRestPagingAction.completeWithPaging(SpotifyRestAction.kt:67)
     at com.adamratzman.spotify.main.SpotifyRestPagingAction.complete(SpotifyRestAction.kt:56)

```

**Comment by dhelleberg** [comment]
Small update: Must be an android thing... I tried the lib as a stand-alone Kotlin application -> works like a charm. Android -> Exception... Is the lib supposed to work on android?

**Comment by dhelleberg** [comment]
My best guess right now: The JSONArray class is part of the android core framework, in non-iterable version. Therefore it doesn't work on android. 
If you want this lib to work on android, it might make sense to use another Json implementation which is not part of android, but -again- this is just a guess.

**Comment by adamint** [comment]
Great, thanks. Later today, the release of `2.0.1` will incorporate this change.

**Comment by dhelleberg** [comment]
big 👍 for the fix, I'll test it once 2.0.1 is out

**Comment by dhelleberg** [comment]
2.0.1 fixes this issue with PR https://github.com/adamint/spotify-web-api-kotlin/pull/33
